src/terrain/clients/analyses.clj

```diff
@@ -0,0 +1,74 @@
+(ns terrain.clients.analyses
+  (:require [terrain.util.config :refer [analyses-base-uri]]
+            [cheshire.core :refer [generate-string]]
+            [clojure.tools.logging :as log]
+            [terrain.util.transformers :refer [add-current-user-to-map]]
+            [clj-http.client :as http]
+            [clojure-commons.client :refer [build-url-with-query]]
+            [clojure-commons.error-codes :refer [ERR_NOT_FOUND
+                                                 ERR_BAD_REQUEST
+                                                 ERR_UNCHECKED_EXCEPTION]]
+            [cemerick.url :refer [url]]
+            [medley.core :refer [mapply]]
+            [slingshot.slingshot :refer [throw+]]))
+
+(defn- fix-user
+  [{:keys [user] :as user-map}]
+  (-> user-map
+    (assoc :user (str user "@iplantcollaborative.org"))
+    (dissoc :first-name :last-name :email)))
+
+(defn analyses-url
+  ([components]
+   (analyses-url components {}))
+  ([components query]
+   (let [user-map (fix-user (add-current-user-to-map query))]
+     (-> (apply url (cons (analyses-base-uri) components))
+        (assoc :query user-map)
+        (str)))))
+
+(defn process-response
+  [resp err-data]
+  (log/warn resp)
+  (cond
+    (= (:status resp) 404)
+    (throw+ (merge {:error_code ERR_NOT_FOUND} err-data {:msg (:body resp)}))
+
+    (= (:status resp) 400)
+    (throw+ (merge {:error_code ERR_BAD_REQUEST} err-data {:msg (:body resp)}))
+
+    (= (:status resp) 500)
+    (throw+ (merge {:error_code ERR_UNCHECKED_EXCEPTION} err-data {:msg (:body resp)}))
+
+    (not (<= 200 (:status resp) 299))
+    (throw+ (merge {:error_code ERR_UNCHECKED_EXCEPTION} err-data {:msg (:body resp)}))
+
+    :else
+    (:body resp)))
```

Comment:
`clj-http` throws exceptions for many HTTP response codes by default. It's possible that this function isn't being used unless the request succeeds. I like the way that you're merging in the error data, though. That's a good idea.

Reply:
I was wondering about that, but saw a similar function elsewhere in the terrain code. I'll try removing the calls and see what happens.

Reply:
This should be fixed now.